src/com/google/cose/Ec2SigningKey.java

```diff
@@ -114,6 +118,74 @@ public ECPublicKey getPublicKey() {
     return (ECPublicKey) this.keyPair.getPublic();
   }
 
+  private static byte[] arrayFromBigNum(BigInteger n, int curveSize) {
+    byte[] rgb = new byte[(curveSize + 7) / 8];
+    byte[] rgb2 = n.toByteArray();
+    if (rgb.length == rgb2.length) {
+      return rgb2;
+    }
+    if (rgb2.length > rgb.length) {
+      System.arraycopy(rgb2, rgb2.length - rgb.length, rgb, 0, rgb.length);
```

Comment:
Should throw if we're truncating anything other than leading 0's.

`IllegalArgumentException`, maybe?

Reply:
Even better: check for only a single leading zero.

Anything more is unexpected.

Reply:
So the purpose of the arrayFromBigNum is to either get rid of padding or add padding,

1. Why is it an issue if the generated byteArray has more than 1 byte of padding?
2.  What's the 1/256 chance that the generated byteArray has 31 bytes? And could it be lower?

Reply:
> Why is it an issue if the generated byteArray has more than 1 byte of padding?

I don't think it's a bug per se, but it's certainly not expected. You end up with padding bytes when a sign bit has been added to indicate unsigned integer. It would be "really weird" if the underlying implementation was adding more padding bytes.

No matter what, this code needs to ensure that the truncated bytes are `0`. IMO it's easier to just check a single byte and error out in other cases. However if you want to allow arbitrary padding lengths then I wouldn't argue the point.

> What's the 1/256 chance that the generated byteArray has 31 bytes? And could it be lower?

This is me likely being very bad at probability. My reasoning is: "What are the chances that the top byte is all 0?". Since that top byte has 256 possible values, and only one of them results in a fully 0 byte, then it's roughly 1/256 chance.

Of course, no leading byte actually requires the top 9/256 bits to be 0. What is the probability of that? I dunno, I'm bad at probability, so I should probably refrain from making anymore silly statements.

Reply:
I'm thinking that if the underlying generation did create a special key(extra/not enough padding) it would be a better idea to handle it and not freak out with an error due to probability. 

I do agree in the case of extra padding that if the truncated bytes were not `0`, that would be reason for error. I'll take your word on the "really weird" and throw an error if there's more than one, as well.